### PR TITLE
Dither performance enhancements

### PR DESCRIPTION
**Pull request by NumbGnat**
Created temporary Dictionary of colors, only including the selected blocks, to reduce the overhead of looping through unselected blocks.

### CODE REVIEW

**Review by EwoutVDC — COMMENTED** [review]
Looking at how the code works, i think we can also just skip the non-selected variants in the constructor instead of when converting.

If that does not work:
I don't like masking the private member averageColors with the member function parameter.
Can you rename that to eg selectedAverageColors?


There's also no need to check for variantColor.Key.Selected in the GetBestVariant loop if we limit the dictionary to selected variants only.

Thanks for your contributions, i hope you have found my work useful!

**Comment by NumbGnat** [comment]
Thanks for the suggestions.  This is much more efficient. 

Yes.  I've found your work useful.  Thanks for putting it out there to share! I wasn't sure if you were even around anymore, because of the gap in work.  But I figured I'd share my updates back up stream in case you happened to be interested.

I was struggling to find a good pixel art generator and ran across yours. I figured, with the source code available, at least I could tweak it if I needed. I've enjoyed spelunking through your code and looking for ways to enhance the app even further.

I've got some more changes Quality of Life improvements soon.

File: MCPixelArtConverter/ImageConverterAverage.cs [hunk 1]
```diff
@@ -19,26 +19,29 @@ public ImageConverterAverage(Dictionary<MCBlockVariant, Bitmap> palette)
         {
             Parallel.ForEach(palette, kv =>
             {
-                ColorDouble averageColor = new ColorDouble(0, 0, 0, 0);
-                Bitmap bm = kv.Value;
-                if (bm == null)
-                    return;
-
-                for (int w = 0; w < bm.Width; w++)
+                if (kv.Key.Selected)
```
Comment (EwoutVDC) [inline MCPixelArtConverter/ImageConverterAverage.cs]: I would write if (!kv.Key.Selected) return;
That prevents nesting one level deeper, which is easier to track

**Comment by EwoutVDC** [comment]
> Thanks for the suggestions. This is much more efficient.
> 
> Yes. I've found your work useful. Thanks for putting it out there to share! I wasn't sure if you were even around anymore, because of the gap in work. But I figured I'd share my updates back up stream in case you happened to be interested.
> 
> I was struggling to find a good pixel art generator and ran across yours. I figured, with the source code available, at least I could tweak it if I needed. I've enjoyed spelunking through your code and looking for ways to enhance the app even further.
> 
> I've got some more changes Quality of Life improvements soon.

Thanks. That's what open source projects are for!
I'm not actively working on the project anymore, but I'll make some time to support issues and pull requests when required.

File: MCPixelArtConverter/ImageConverterAverage.cs (hunk 1 above)
Comment (NumbGnat) [reply MCPixelArtConverter/ImageConverterAverage.cs]: Done.

**Pull request merged**